tokio/src/runtime/task/mod.rs

```diff
@@ -82,33 +75,16 @@ pub(crate) trait Schedule: Sync + Sized + 'static {
 }
 
 cfg_rt! {
-    /// Create a new task with an associated join handle
-    pub(crate) fn joinable<T, S>(task: T) -> (Notified<S>, JoinHandle<T::Output>)
-    where
-        T: Future + Send + 'static,
-        S: Schedule,
-    {
-        let raw = RawTask::new::<_, S>(task);
-
-        let task = Task {
-            raw,
-            _p: PhantomData,
-        };
-
-        let join = JoinHandle::new(raw);
-
-        (Notified(task), join)
-    }
-}
-
-cfg_rt! {
-    /// Create a new `!Send` task with an associated join handle
-    pub(crate) unsafe fn joinable_local<T, S>(task: T) -> (Notified<S>, JoinHandle<T::Output>)
+    /// Create a new task with an associated join handle. This method is used
+    /// only for blocking tasks and tests. Other spawned tasks use the bind
+    /// method on OwnedTasks.
+    pub(crate) fn joinable<T, S>(task: T, scheduler: S) -> (Notified<S>, JoinHandle<T::Output>)
     where
-        T: Future + 'static,
+        T: Send + Future + 'static,
         S: Schedule,
     {
-        let raw = RawTask::new::<_, S>(task);
+        let raw = RawTask::new::<_, S>(task, scheduler);
+        raw.header().state.ref_dec();
```

Comment:
We should add a comment explaining what this `ref_dec()` is for. My guess is it is because there is no task set owning this task?

I would *probably* rename the `joinable` function to something like `unowned(...)` or something like that to make it more obvious what the fn is for now.